### PR TITLE
Reorganise the Logging directory

### PR DESCRIPTION
**Pull request by ianpartridge**
This is just moving code around so everything isn't in one big file. Also, add copyright header (let me know if a different format is needed).

### CODE REVIEW

File: Sources/Logging/Logging.swift [hunk 1]
```diff
@@ -1,128 +1,40 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
 
 // This is the logging system itself, it's mostly used to obtain loggers and to set the type of the `LogHandler`
 // implementation.
 public enum Logging {
     private static let lock = ReadWriteLock()
-    private static var _factory: (String) -> LogHandler = StdoutLogger.init
+    private static var factory: (String) -> LogHandler = StdoutLogHandler.init
```
Comment (ianpartridge) [inline Sources/Logging/Logging.swift:19]: These renames seemed sensible to me.

File: Sources/Logging/Logging.swift [hunk 2]
```diff
@@ -138,138 +50,19 @@ extension Logging.Level: Comparable {
...
-    public init(_ factories: [(String) -> LogHandler]) {
-        self.factories = factories
-    }
-
-    public func make(label: String) -> LogHandler {
-        return MUXLogHandler(handlers: self.factories.map { $0(label) })
-    }
-}
-
-private class MUXLogHandler: LogHandler {
```
Comment (ianpartridge) [inline Sources/Logging/Logging.swift:156]: Renamed this to `MultiplexLogHandler`.

File: Sources/Logging/Logging.swift [hunk 3]
```diff
@@ -1,128 +1,40 @@
...
-    public var logLevel: Logging.Level {
-        get {
-            return self.handler.logLevel
-        }
-        set {
-            self.handler.logLevel = newValue
-        }
-    }
-}
+//===----------------------------------------------------------------------===//
```
Comment (weissi) [inline Sources/Logging/Logging.swift:1]: @ianpartridge this adds an empty file with just a license header

Comment (ianpartridge) [reply Sources/Logging/Logging.swift:1]: I think `public enum Logging` is still in here?

File: Sources/Logging/Logging.swift (hunk 2 above)
Comment (ktoso) [reply Sources/Logging/Logging.swift:156]: Hm, if so then `Multiplexing` sounds more correct? `MultiplexingLogHandler`, WDYT?

**Comment by ianpartridge** [comment]
Closing this as it has conflicts already. We can split them out when things have settling down a bit?

**Comment by weissi** [comment]
@ianpartridge yes, we'll clear this up and add the license. The examples will also be removed etc.

**Pull request closed without merging**